session.go

```diff
@@ -695,8 +700,14 @@ func (addr *ServerAddr) String() string {
 }
 
 // TCPAddr returns the resolved TCP address for the server.
+// TCPAddr will panic if ServerAddr is a Unix Domain Socket
 func (addr *ServerAddr) TCPAddr() *net.TCPAddr {
-	return addr.tcp
+	return (*net.TCPAddr)(addr.addr.(*net.TCPAddr))
+}
```

Comment:
What if addr is not actually a net.TCPAddr?

Reply:
It panics. That's a documented behavior.
Same problem exists before my PR.

Reply:
Right, sorry, I didn't see the comment just above this, the PR review UI (in)conveniently hides that.